code/modules/mapping/mapping_helpers.dm

```diff
@@ -546,6 +546,27 @@
 		log_mapping("[src] at [AREACOORD(src)] [(apc_area.type)] tried to set [target]'s charge to 100 but it's already at 100!")
 	target.full_charge = TRUE
 
+//Used to turn off lights in areas.
+/obj/effect/mapping_helpers/turn_off_lights
+	name = "area lights off helper"
+	icon_state = "lights_off"
+	late = TRUE
+
+/obj/effect/mapping_helpers/turn_off_lights/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		log_mapping("[src] spawned outside of mapload!")
+		return INITIALIZE_HINT_QDEL
+	payload()
+	qdel(src)
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/turn_off_lights/proc/payload()
+	var/area/needed_area = get_area(src)
+	if(!needed_area.lightswitch)
+		log_mapping("[src] at [AREACOORD(src)] [(needed_area.type)] tried to turn lights off but they are already off!")
+	needed_area.lightswitch = FALSE
```

Comment:
Should this be a stacktrace?

Reply:
What's stacktrace?

Reply:
it logs a bit of text as an error like CRASH, but it allows the proc to continue execution.
It's good for doing something suspect under protest. You can raise a warning and then do the thing anyway, on the off chance it's the desired outcome

Reply:
I'm not sure if this is a good idea tho